js/models/contact_model.js

```diff
@@ -9,22 +9,24 @@ angular.module('contactsApp')
 			addressBookId: addressBook.displayName,
 
 			uid: function(value) {
+				var model = this;
 				if (angular.isDefined(value)) {
 					// setter
-					return this.setProperty('uid', { value: value });
+					return model.setProperty('uid', { value: value });
 				} else {
 					// getter
-					return this.getProperty('uid').value;
+					return model.getProperty('uid').value;
 				}
 			},
 
 			fullName: function(value) {
+				var model = this;
```

Comment:
because otherwise I'm getting a warning in my ide

'Potentially invalid usage of this'